extension/js/service_worker/bg-handlers.ts

```diff
@@ -112,4 +113,66 @@ export class BgHandlers {
         }
       });
     });
+
+  public static thunderbirdSecureComposeHandler = () => {
+    const handleClickEvent = async (tabId: number, acctEmail: string, thunderbirdMsgId: number, composeMethod?: messenger.compose._ComposeDetailsType) => {
+      const accountEmails = await GlobalStore.acctEmailsGet();
+      const useFullScreenSecureCompose = (await messenger.windows.getCurrent()).type === 'messageCompose';
+      composeMethod = composeMethod === 'reply' || composeMethod === 'forward' ? composeMethod : undefined;
+      if (accountEmails.length !== 0) {
+        await BgUtils.openExtensionTab(
+          Url.create('/chrome/settings/inbox/inbox.htm', { acctEmail, useFullScreenSecureCompose, thunderbirdMsgId, composeMethod })
+        );
+        await messenger.tabs.remove(tabId);
+      } else {
+        await BgUtils.openExtensionTab(Url.create('/chrome/settings/initial.htm', {}));
+      }
+    };
+    messenger.composeAction.onClicked.addListener(async tab => {
+      const messageDetails = await messenger.compose.getComposeDetails(Number(tab.id));
+      const composeMethod = messageDetails.type;
+      const msgId = Number(messageDetails.relatedMessageId);
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const acctEmail = Str.parseEmail(messageDetails.from as string).email!;
+      await handleClickEvent(Number(tab.id), acctEmail, msgId, composeMethod);
+    });
+    messenger.messageDisplayAction.onClicked.addListener(async tab => {
+      const tabId = Number(tab.id);
+      const messageDetails = await messenger.messageDisplay.getDisplayedMessage(tabId);
+      if (messageDetails) {
+        const msgId = messageDetails.id;
+        const accountId = messageDetails?.folder?.accountId || '';
+        const acctEmail = (await messenger.accounts.get(accountId))?.name || '';
+        await handleClickEvent(tabId, acctEmail, msgId);
+      }
+    });
+  };
+
+  public static thunderbirdContentScriptRegistration = async () => {
+    await messenger.messageDisplayScripts.register({
+      js: [{ file: './js/content_scripts/thunderbird-content-script.js' }],
```

Comment:
after build changes all extension versions (not only thunderbird), include additional `thunderbird-content-script.js` which is almost 5mb:

<img width="669" alt="Screenshot 2024-08-26 at 20 16 37" src="https://github.com/user-attachments/assets/2798fb35-3b88-4d5b-84bc-7f12e95e4f08">

for browsers we use [injectFcIntoWebmail](https://github.com/FlowCrypt/flowcrypt-browser/blob/a84810e6d1598d1a38a07881a1a98d56427a779f/extension/js/service_worker/inject.ts#L7) method which adds all needed files to each gmail tab.

do you think it'll be possible to adapt this method for thunderbird?

Reply:
> after build changes all extension versions (not only thunderbird), include additional thunderbird-content-script.js which is almost 5mb:

Ah, yes - sorry about that. I'll find a way a viable alternative solution to it from which `thunderbird-content-script.js` should only be created for the Thunderbird build.

Reply:
let's start with trying to adapt [injectFcIntoWebmail](https://github.com/FlowCrypt/flowcrypt-browser/blob/a84810e6d1598d1a38a07881a1a98d56427a779f/extension/js/service_worker/inject.ts#L7) method to inject all needed scripts into thunderbird, so it'll work the same in browsers and thunderbird.
and if it doesn't work - then use `thunderbird-content-script.js`

Reply:
Hi @sosnovsky - I tried to adapt using injectFcIntoWebmail and there are incompatibility that prevents me from somehow make it work as close as how it works on browser. For example, in Thunderbird, registration for content script is done within `messageDisplayScript.register`, from which it only accept one of the two parameter (file or code). And then several files doesnt seem to work when defined in `.register()` - https://webextension-api.thunderbird.net/en/91/messageDisplayScripts.html.

However, I have done my best to achieve the best result while still complying with the requirements, which I believe I have accomplished in this PR. Please review and test it out. From my perspective, the results are extremely positive, and I'm excited to hear your feedback. :)

Thanks for your help.